alibi/explainers/shap_wrappers.py

```diff
@@ -207,6 +209,57 @@ def _reduction(arr, axis, indices=None):
     return np.add.reduceat(values, slices, axis=1)
 
 
+DISTRIBUTED_OPTS = {
+    'n_cpus': None,
+    'batch_size': None,
+}
+
+
+class KernelExplainerWrapper(KernelExplainer):
+    """
+    A wrapper around `shap.KernelExplainer` that supports:
+
+        - fixing the seed when instantiating the KernelExplainer in a separate process
+        - passing a batch index to the explainer so that a parallel explainer pool can return batches in arbitrary order
+    """
+    def __init__(self, *args, **kwargs):
+        seed = None
+        if 'seed' in kwargs:
+            seed = kwargs.pop('seed')
```

Comment:
Why not simply `seed=kwargs.get('seed', None)` ?

Reply:
`seed` is part of the kwargs only in a distributed context, and it is used to control the randomness inside the `shap` library from every process. Two things I overlooked:

- Does setting the seed before invoking the distributed context "carry over" to each process (eg, is the call inside `fit` sufficient to set the random stream across all processes?) 
- L229 has an undesired side-effect of resetting the seed set in `fit` (breaks non-distributed version) - to be fixed.